src/components/navigation/NavigationKeyHandler.ts

```diff
@@ -0,0 +1,285 @@
+/**
```

Comment:
```suggestion
import { wait } from "@feedzai/js-utilities";
import { wiper } from "../../utils/wiper";
import { addDataNavigation } from "./AddNavigation";
import { jumpXCharts, jumpXElements } from "./JumpX";
import { skip } from "./Skip";
import { xSetter } from "./XSetter";

interface NavigationKeyHandlerParams {
  type: string;
  event: React.KeyboardEvent;
  number: number;
  chartRef: React.RefObject<HTMLElement>;
  elements: HTMLElement[];
  alertDivRef: React.RefObject<HTMLElement>;
  selectedSeries: string;
  series: string[];
  selectorType: { element?: string; className?: string };
  multiSeries?: string;
  setSelectedSeries?: (series: string) => void;
}

interface SwitchSeriesParams {
  chartRef: React.RefObject<HTMLElement>;
  selectorType: { element?: string; className?: string };
  selectedSeries: string;
  series: string[];
}

const MODAL_CONTENT_CLASS = 'a11y_modal_content';
const ROW_CLASS = 'a11y_row';
const GUIDE_CLOSE_ID = 'guide_close';
const DESC_CLASS = 'a11y_desc';

const isModalElement = (element: Element | null): boolean =>
  element?.classList.contains(MODAL_CONTENT_CLASS) ||
  element?.classList.contains(ROW_CLASS) ||
  element?.id === GUIDE_CLOSE_ID;

/**
 * Handles navigation key events for chart interaction.
 * @param {NavigationKeyHandlerParams} params - The parameters for navigation key handling.
 * @returns {number} The updated number of points to jump.
 */
export function navigationKeyHandler(params: NavigationKeyHandlerParams): number {
  const { type, event, number, chartRef, elements, alertDivRef, selectedSeries, series, selectorType, multiSeries, setSelectedSeries } = params;
  const { altKey, key, code } = event.nativeEvent;

  try {
    const updatedNumber = xSetter({ event, type, number, alertDivRef });
    skip({ event, chartRef, selectorType, selectedSeries });

    const charts = Array.from(document.getElementsByClassName(DESC_CLASS));
    const chart = chartRef.current?.getElementsByClassName(DESC_CLASS)[0] as HTMLElement;
    
    if (chart === document.activeElement && charts.includes(chart)) {
      jumpXCharts({ event, charts, chart });
    } else {
      jumpXElements({ event, number: updatedNumber, elements, selectedSeries, series });
    }

    if (altKey && code === "KeyM") {
      event.preventDefault();
      return handleAltM({ number: updatedNumber, selectedSeries, series, selectorType, setSelectedSeries, chartRef, multiSeries, alertDivRef });
    }

    switch (key) {
      case "ArrowDown":
        handleArrowDown(event, chartRef, selectorType, selectedSeries, alertDivRef);
        break;
      case "ArrowUp":
        handleArrowUp(event, chartRef, alertDivRef);
        break;
      case "?":
        handleQuestionMark(event, chartRef);
        break;
    }

    return updatedNumber;
  } catch (error) {
    console.error('Error in navigationKeyHandler:', error);
    return number;
  }
}

function handleAltM(params: Omit<NavigationKeyHandlerParams, 'type' | 'event' | 'elements'>): number {
  const { number, selectedSeries, series, selectorType, setSelectedSeries, chartRef, multiSeries, alertDivRef } = params;

  if (isModalElement(document.activeElement)) return number;
  
  if (document.activeElement?.classList.contains(DESC_CLASS)) {
    showAlert(alertDivRef, "You can only change series while focused on a data point");
    return number;
  }
  
  if (!multiSeries) {
    showAlert(alertDivRef, "This chart only has one series of data");
    return number;
  } 
  
  if (setSelectedSeries && selectorType && selectedSeries && series.length > 0) {
    const currentPos = series.indexOf(selectedSeries);
    const nextPos = (currentPos + 1) % series.length;
    setSelectedSeries(series[nextPos]);
    switchSeries({ chartRef, selectorType, selectedSeries, series });
  }
  
  return number;
}

function handleArrowDown(event: React.KeyboardEvent, chartRef: React.RefObject<HTMLElement>, selectorType: { element?: string; className?: string }, selectedSeries: string, alertDivRef: React.RefObject<HTMLElement>): void {
  event.preventDefault();
  if (isModalElement(document.activeElement)) return;
  if (!document.activeElement?.classList.contains(DESC_CLASS)) {
    showAlert(alertDivRef, "You are already at the data level");
    return;
  }
  switchToDataLevel({ chartRef, selectorType, selectedSeries });
}

function handleArrowUp(event: React.KeyboardEvent, chartRef: React.RefObject<HTMLElement>, alertDivRef: React.RefObject<HTMLElement>): void {
  event.preventDefault();
  if (isModalElement(document.activeElement)) return;
  if (document.activeElement?.classList.contains(DESC_CLASS)) {
    showAlert(alertDivRef, "You are already at the chart level");
    return;
  }
  switchToChartLevel(chartRef);
}

function handleQuestionMark(event: React.KeyboardEvent, chartRef: React.RefObject<HTMLElement>): void {
  const modal = document.getElementsByClassName("a11y_modal")[0];
  if (modal) {
    event.preventDefault();
    levelGuide(chartRef);
  }
}

async function showAlert(alertDivRef: React.RefObject<HTMLElement>, message: string): Promise<void> {
  if (alertDivRef.current) {
    alertDivRef.current.textContent = message;
    await wait(1000);
    alertDivRef.current.textContent = "\u00A0";
  }
}

function switchSeries({ chartRef, selectorType, selectedSeries, series }: SwitchSeriesParams): void {
  const previousPoint = document.activeElement as HTMLElement;
  const elements = getElements(chartRef, selectorType);
  
  const previousSeries = elements?.filter(element => 
    element.classList.contains(`series:${selectedSeries}`)
  ) ?? [];

  const previousIndex = previousSeries.indexOf(previousPoint);
  const previousSeriesPos = series.indexOf(selectedSeries);
  const currentSeriesPos = (previousSeriesPos + 1) % series.length;
  const currentSeriesName = series[currentSeriesPos].replace(/ /g, "-");

  const currentSeries = elements?.filter(element => 
    element.classList.contains(`series:${currentSeriesName}`)
  ) ?? [];

  wiper(chartRef);

  addDataNavigation({
    chartRef,
    selectorType,
    selectedSeries: currentSeriesName,
    focusPoint: currentSeries[previousIndex],
  });
}

function getElements(chartRef: React.RefObject<HTMLElement>, selectorType: { element?: string; className?: string }): HTMLElement[] {
  if (selectorType.element && chartRef.current) {
    return Array.from(chartRef.current.querySelectorAll(selectorType.element));
  } else if (selectorType.className && chartRef.current) {
    return Array.from(chartRef.current.getElementsByClassName(selectorType.className));
  }
  return [];
}
```
Key improvements:
- Better error handling: Added a try-catch block in the main function.
- Modularization: Broke down the large navigationKeyHandler function into smaller, more manageable functions.
- Type safety: Improved type annotations and made use of TypeScript's type inference capabilities.
- Null checks: Added more null checks to prevent potential runtime errors.
- Consistency: Made the code style more consistent throughout the file.
- Documentation: Added JSDoc comments for the main function.
- Removed unnecessary type assertions where possible.
- Simplified some of the logic and removed redundant checks.